registry/registry.go

```diff
@@ -307,23 +308,27 @@ func (registry *Registry) ListenAndServe() error {
 		dcontext.GetLogger(registry.app).Infof("listening on %v", ln.Addr())
 	}
 
-	if config.HTTP.DrainTimeout == 0 {
-		return registry.server.Serve(ln)
-	}
-
 	// setup channel to get notified on SIGTERM signal
-	signal.Notify(registry.quit, syscall.SIGTERM)
+	signal.Notify(registry.quit, os.Interrupt, syscall.SIGTERM)
 	serveErr := make(chan error)
 
 	// Start serving in goroutine and listen for stop signal in main thread
 	go func() {
 		serveErr <- registry.server.Serve(ln)
 	}()
+	defer func(app *handlers.App) {
```

Comment:
Can we now move this into registry.Shutdown so its with other shutdown actions?

I expect if an embedded registry is shutdown, closing this would also be appropriate.